src/main/java/me/coley/recaf/util/RegexUtil.java

```diff
@@ -101,11 +104,33 @@ public static Matcher getMatcher(String pattern, String text) {
 	 *
 	 * @return Matched words.
 	 */
-	public static String[] matches(String text, String pattern) {
+	public static String[] allMatches(String text, String pattern) {
 		List<String> list = new ArrayList<>();
 		Matcher m = getMatcher(pattern, text);
 		while(m.find())
 			list.add(m.group(0));
 		return list.toArray(EMPTY);
 	}
+
+	/**
+	 * Creates new {@link Pattern} or gets it from cache.
+	 *
+	 * @param regex pattern's regex
+	 * @return {@link Pattern}
+	 */
+	public static Pattern pattern(String regex) {
+		return PATTERNS.computeIfAbsent(regex, Pattern::new);
+	}
+
+	/**
+	 * Checks if pattern matches an input.
```

Comment:
Should be `Checks if the entire input matches a pattern`

Since this is basically how `String.matches()` works, this checks if the entire length of the string matches. As opposed to the method above which will yield substring matches of the given text.